Scripts/Runtime/Data/Collections/UnsafeHeap.cs

```diff
@@ -0,0 +1,610 @@
+using Anvil.Unity.DOTS.Entities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    //NOTE - Taken from: https://github.com/Amarcolina/NativeHeap (MIT LICENCE)
```

Comment:
If they already have unit tests we should steal those too.

Reply:
Done